ocp_utilities/infra.py

```diff
@@ -81,6 +85,73 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(
+    nodes,
+    bypass_check_if_kubelet_ready=True,
+    healthy_node_condition_type=None,
+):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+         bypass_check_if_kubelet_ready:
```

Comment:
`bypass_check_if_kubelet_ready` we do not need this, the user can send the desire checks in the dict.

Reply:
@myakove According to discussion with @dbasunag, this new function will break existing testsuites. 
> "... Current code only check for node ready, your change that would be introducing all the default conditions, which would definitely not work for us. "

The `bypass_check_if_kubelet_ready` defaulting to `True` will not `cause cluster_sanity` to  Raise an exception, if the **Disk** is degraded, but `node.kubelet_ready` is `True`.
https://github.com/RedHatQE/openshift-python-utilities/pull/113#discussion_r1093444655

> [@rnetser](https://github.com/rnetser) we should have an option to bypass this check. A node with disk/pid/memory pressure, not necessarily indicates that the cluster is unhealthy or tests should be stopped to triage. It could be due to temporary issues - e.g. often seen during upgrade tests or if we are updating api server and so on. As long as the kubelet is in ready state, we should not fail cluster sanity.

Reply:
Once you add a dict the user can provide the desire dict so CNV can send what they needed to the function

Reply:
done. removed bypass_check_if_kubelet_ready